app/app/src/main/kotlin/com/hedvig/app/feature/genericauth/SsnInputScreen.kt

```diff
@@ -0,0 +1,176 @@
+package com.hedvig.app.feature.genericauth
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.safeDrawingPadding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.unit.dp
+import com.hedvig.android.core.designsystem.component.button.HedvigContainedButton
+import com.hedvig.android.core.designsystem.component.textfield.HedvigTextField
+import com.hedvig.android.core.designsystem.preview.HedvigPreview
+import com.hedvig.android.core.designsystem.theme.HedvigTheme
+import com.hedvig.android.core.ui.appbar.TopAppBarWithBack
+import com.hedvig.android.market.Market
+import hedvig.resources.R
+
+@Composable
+fun SSNInputScreen(
+  market: Market,
+  onUpClick: () -> Unit,
+  onInputChanged: (String) -> Unit,
+  onSubmitEmail: () -> Unit,
+  emailInput: String,
+  error: String?,
+  loading: Boolean,
+) {
+  Column(
+    modifier = Modifier
+      .safeDrawingPadding()
+      .fillMaxSize(),
+  ) {
+    TopAppBarWithBack(
+      onClick = onUpClick,
+      title = stringResource(R.string.zignsec_login_screen_title),
+    )
+    Column(
+      modifier = Modifier
+        .fillMaxSize()
+        .padding(horizontal = 16.dp)
+        .verticalScroll(rememberScrollState()),
+    ) {
+      Spacer(Modifier.height(60.dp))
+      Text(
+        text = stringResource(R.string.zignsec_login_screen_title),
+        style = MaterialTheme.typography.headlineMedium,
+      )
+      Spacer(Modifier.height(20.dp))
+      SSNTextField(
+        input = emailInput,
+        market = market,
+        onInputChanged = onInputChanged,
+        onSubmit = onSubmitEmail,
```

Comment:
We call `onSubitEmail` with an `ssn` here as I understand it. Is this WAI and is the name just a bit confusing? If so we might wanna rename so we do not fall into wrong assumptions when we see `onSubmitEmail` higher up and assume it's actually an email.